### ISSUE TITLE
Smithy to Typescript union not generating members correctly

### ISSUE THREAD

**Issue by joelvanderhoof** [aws/aws-pdk#333]
### Describe the bug

When generating a Typescript client with `SmithyApiGatewayTsProject`, structures that are members of unions from the Smithy model are generated as Typescript interfaces with a single property that is the name of the interface.  The traits from the Smithy structure are not included.

### Expected Behavior

When a Smithy structure is a member of a union, the traits of the structure should be included in the Typescript interface that is generated as a member of the Typescript union.

### Current Behavior

The Smithy structure that is a member of a Smithy union generates a Typescript interface that has a single property that is the interface name. The Typescript interface does not have any properties corresponding to the Smithy structure traits.

### Reproduction Steps

main.smithy
```
use aws.protocols#restJson1

/// A sample smithy api
@restJson1
service api {
    version: "1.0"
    operations: [Foo]
}


// Foo test route ---------------------------------------------------
@readonly
@http(method: "GET", uri: "/foo/{something}")
operation Foo {
    input: FooInput
    output: FooOutput
    errors: [ApiError]
}

@input
structure FooInput {
    @required
    @httpLabel
    something: String
}


@output
structure FooOutput {
    foo: OutputsList
}

list OutputsList {
    member: Bar
}

union Bar {
    baz: Baz
    qux: Qux
}

structure Baz {
    type: String
    day: Document
}

structure Qux {
    order: Integer
    description: String
}
```

Generated Typescript files:

Bar.ts
```
import {
    Baz,
    instanceOfBaz,
    BazFromJSON,
    BazFromJSONTyped,
    BazToJSON,
} from './Baz';
import {
    Qux,
    instanceOfQux,
    QuxFromJSON,
    QuxFromJSONTyped,
    QuxToJSON,
} from './Qux';

/**
 * @type Bar
 * 
 * @export
 */
export type Bar = Baz | Qux;
```

Baz.ts: Note that the `Baz` interface does not have any of the traits of the `Baz` structure, only a property 'baz' that references itself.
```
export interface Baz {
    /**
     * 
     * @type {Baz}
     * @memberof Baz
     */
    baz: Baz;
}
```

Qux.ts: Note that the `Qux` interface does not have any of the traits of the `Qux` structure, only a property 'qux' that references itself.
```
export interface Qux {
    /**
     * 
     * @type {Qux}
     * @memberof Qux
     */
    qux: Qux;
}
```

### PDK version used

0.14.20

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

MacOS 13.2.1

**Comment by cogwirrel** [comment]
Nice find!

As a workaround, the generated code seems correct if the union properties are named something different to their types, eg:

```smithy
union Bar {
  baz: BazType
  qux: QuxType
}
```

Then becomes:

```ts
export interface Baz {
  baz: BazType;
}

export interface Qux {
  qux: QuxType;
}
```

Not ideal, as it's often most intuitive to name them the same, so there's definitely some further investigation required to find the root cause, but hopefully this unblocks you! :)

**Comment by joelvanderhoof** [comment]
Yep! That works! Thanks so much for the quick response!

**Comment by cogwirrel** [comment]
Awesome! No worries! :) I might leave this open at least until I find (or raise) the relevant issue in open api generator :)

**Comment by cogwirrel** [comment]
Closing this in favour of the issue I raised with Open API Generator here: https://github.com/OpenAPITools/openapi-generator/issues/15212
